src/Nancy.Hosting.Self/NancyHost.cs

```diff
@@ -124,20 +125,27 @@ public void Start()
         {
             this.StartListener();
 
-            Task.Factory.StartNew(async () =>
+            Task.Run(() =>
             {
-                try
+                var semaphore = new Semaphore(this.configuration.MaximumConnectionCount, this.configuration.MaximumConnectionCount);

```

Comment:
I'm certainly open to using slim, however the way I understand slim is that it's intended for use when wait times are expected to be short. I'm not sure that there's a gaurantee for that to be true in all cases, especially for a service that experiences no load during certain times of the day. The wait time in this context is the time between a WaitOne a call and GetContextAsync completing and firing the continuation. What do you think?